src/webloader.js

```diff
@@ -53,32 +53,25 @@ function injectStylesheet(endpoint, env, version, uid) {
 	document.head.appendChild(stylesheet);
 }
 
-function handlePushstate(client) {
+function handlePushState(client) {
 	const pushStateOrig = history.pushState;
+	const replaceStateOrig = history.replaceState;
 
-	history.pushState = function() {
-		const args = Array.prototype.slice.call(arguments);
-		pushStateOrig.apply(history, args);
-	
-		let event;
-		const eventType = 'pushstate_evolv';
-	
-		if (Event.prototype.constructor) {
-			event = new CustomEvent(eventType, {});
-		} else { // For IE Compatibility
-			event = document.createEvent('Event');
-			event.initEvent(eventType);
-		}
-	
-		window.dispatchEvent(event);
-	};
-
-	const updateContext = function () {
+	function updateContext() {
 		client.context.set('web.url', window.location.href);
-	};
+	}
+
+	function handler(orig) {
+		const args = Array.prototype.slice.call(arguments, 1);
+		orig.apply(history, args);
+
+		updateContext();
+	}
+
+	history.pushState = handler.bind(this, pushStateOrig);
+	history.replaceState = handler.bind(this, replaceStateOrig);
 
 	window.addEventListener('popstate', updateContext);
```

Comment:
Would it make sense to eventually do the same surgery to updateContext on history.back, history.forward, and history.go? Or are there other APIs that send the popstate event that we'll always have to be listening for?

Reply:
The popstate is the safer option, we're only doing this monkey patching because that is unavailable for these 2. Hopefully at some point in the future they fix this part of the spec. Monkey patching should always be a last resort.